src/main/java/de/rototor/pdfbox/graphics2d/PdfBoxGraphics2D.java

```diff
@@ -804,4 +845,14 @@ public FontRenderContext getFontRenderContext() {
 		calcGfx.addRenderingHints(renderingHints);
 		return calcGfx.getFontRenderContext();
 	}
+
+	// In case a single 2D graphic object has to be written (one or more times)
+	// into a PDF document, this method isn't needed.
+	// If, instead, several 2D graphic objects have to be written (one or more times)
+	// into a PDF document, the graphic state has to be saved after a graphic object
+	// has been written for the last time and a new graphic object has to be created
+	public void saveGraphicsState() throws IOException
```

Comment:
Thank you for your reply.

I'm afraid now I'm not able to create the testcase, as this happens in a bigger environment with thousands of classes.
I will try to build something from scratch, hoping it will behave the same way. Maybe next week